File: src/types/syncIndex.ts
export type SyncIndexItemStatus =
  | "up-to-date"
  | "update-available"
  | "not-synced"
  | "failed";

export type SyncIndexItem = {
  appId: string;
  sourceKey: string;
  providerId: string;
  providerName: string;
  fileType: string;
  installedPath: string;
  lastDownloadUrl: string;
  remoteHash?: string;
  localHash?: string;
  etag?: string;
  lastModified?: string;
  installedAt: string;
  updatedAt: string;
  lastCheckedAt?: string;
  status: SyncIndexItemStatus;
};

export type SyncIndex = {
  version: number;
  items: Record<string, SyncIndexItem>;
};

export type SyncCheckResult = {
  appId: string;
  status: string;
  hasUpdate: boolean;
  message: string;
};

File: src-tauri/src/models/sync_index.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct SyncIndexItem {
    pub app_id: String,
    pub source_key: String,
    pub provider_id: String,
    pub provider_name: String,
    pub file_type: String,
    pub installed_path: String,
    pub last_download_url: String,
    pub remote_hash: Option<String>,
    pub local_hash: Option<String>,
    pub etag: Option<String>,
    pub last_modified: Option<String>,
    pub installed_at: String,
    pub updated_at: String,
    pub last_checked_at: Option<String>,
    pub status: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct SyncIndex {
    pub version: u32,
    pub items: std::collections::HashMap<String, SyncIndexItem>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct SyncCheckResult {
    pub app_id: String,
    pub status: String,
    pub has_update: bool,
    pub message: String,
}
